lib/purgatory/purgatory.rb

```diff
@@ -47,6 +47,7 @@ def approve!(approver = nil)
     return false if approved? 
     if soul_with_changes.save
       soul.send(performable_method[:method],*performable_method[:args]) if performable_method
+      return false if soul.errors.any?
```

Comment:
Hmm I don't understand why this is needed. `soul_with_changes.save` succeeded, so why would `soul` have errors?